tests/models/pop2piano/test_feature_extraction_pop2piano.py

```diff
@@ -0,0 +1,342 @@
+            dtype=np.float32,
+        )
+        speech_input2 = np.ones(
+            [
+                2_000_000,
+            ],
+            dtype=np.float32,
+        )
+        speech_input3 = np.random.randint(low=0, high=10, size=500_000)
+        speech_input3 = speech_input3.astype(np.float32).reshape(
+            -1,
+        )
+
+        input_features = feature_extractor(
+            [speech_input1, speech_input2, speech_input3], sampling_rate=[44_100, 16_000, 48_000], return_tensors="np"
+        )
+
+        # check np array or not
+        self.assertEqual(type(input_features["input_features"]), np.ndarray)
+
+        # check shape
+        self.assertEqual(len(input_features["input_features"].shape), 3)
+
+    def test_batch_feature_pt(self):
+        feature_extractor = self.feature_extraction_class(**self.feat_extract_tester.prepare_feat_extract_dict())
+        speech_input1 = np.zeros(
+            [
+                1_000_000,
+            ],
+            dtype=np.float32,
+        )
+        speech_input2 = np.ones(
+            [
+                2_000_000,
+            ],
+            dtype=np.float32,
+        )
+        speech_input3 = np.random.randint(low=0, high=10, size=500_000)
+        speech_input3 = speech_input3.astype(np.float32).reshape(
+            -1,
+        )
+
+        input_features = feature_extractor(
+            [speech_input1, speech_input2, speech_input3], sampling_rate=[44_100, 16_000, 48_000], return_tensors="pt"
+        )
+
+        # check pt tensor or not
+        self.assertEqual(type(input_features["input_features"]), torch.Tensor)
+
+        # check shape
+        self.assertEqual(len(input_features["input_features"].shape), 3)
+
+    @require_tf
+    def test_batch_feature_tf(self):
+        import tensorflow as tf
+
+        feature_extractor = self.feature_extraction_class(**self.feat_extract_tester.prepare_feat_extract_dict())
+        speech_input1 = np.zeros(
+            [
+                1_000_000,
+            ],
+            dtype=np.float32,
+        )
+        speech_input2 = np.ones(
+            [
+                2_000_000,
+            ],
+            dtype=np.float32,
+        )
+        speech_input3 = np.random.randint(low=0, high=10, size=500_000)
+        speech_input3 = speech_input3.astype(np.float32).reshape(
+            -1,
+        )
+
+        input_features = feature_extractor(
+            [speech_input1, speech_input2, speech_input3], sampling_rate=[44_100, 16_000, 48_000], return_tensors="tf"
+        )
+
+        # check tf tensor or not
+        self.assertTrue(tf.is_tensor(input_features["input_features"]))
+
+        # check shape
+        self.assertEqual(len(input_features["input_features"].shape), 3)
+
+    @unittest.skip(
+        "Pop2PianoFeatureExtractor does not supports padding externally(while processing audios in batches padding is automatically applied to max_length)"
+    )
+    def test_padding_accepts_tensors_pt(self):
```

Comment:
Note that these padding tests would work if we used `self.pad` in the feature extractor